src/server/editorServices.ts

```diff
@@ -855,65 +858,247 @@ namespace ts.server {
-                    return { configFileName, configFileErrors: errors };
+        private closeWatchConfigFileForInferredProjectRoot(configFileName: NormalizedPath, canonicalConfigFilePath: string, info: ScriptInfo, reason: WatcherCloseReason) {
+            const configFilePresenceInfo = this.mapOfConfigFilePresence.get(canonicalConfigFilePath);
+            if (configFilePresenceInfo) {
+                // Set this as not inferred root
+                if (configFilePresenceInfo.trackingOpenFilesMap.has(info.path)) {
+                    configFilePresenceInfo.trackingOpenFilesMap.set(info.path, false);
+                    this.logConfigFileWatchUpdate(configFileName, configFilePresenceInfo, ConfigFileWatcherStatus.InferredRootRemoved);
                 }
 
-                // even if opening config file was successful, it could still
-                // contain errors that were tolerated.
-                this.logger.info(`Opened configuration file ${configFileName}`);
-                if (errors && errors.length > 0) {
-                    return { configFileName, configFileErrors: errors };
-                }
-            }
-            else {
-                this.updateConfiguredProject(project);
+                // Close the watcher if present
+                this.closeConfigFileWatcherIfInferredRoot(configFileName, canonicalConfigFilePath,
+                    configFilePresenceInfo, /*infoIsInferredRoot*/ true, reason
+                );
             }
+        }
 
-            return { configFileName };
+        /**
+         * This is called by inferred project whenever root script info is removed from it
+         */
+        /* @internal */
+        stopWatchingConfigFilesForInferredProjectRoot(info: ScriptInfo, reason: WatcherCloseReason) {
+            this.enumerateConfigFileLocations(info, (configFileName, canonicalConfigFilePath) =>
+                this.closeWatchConfigFileForInferredProjectRoot(configFileName, canonicalConfigFilePath, info, reason)
+            );
         }
 
-        // This is different from the method the compiler uses because
-        // the compiler can assume it will always start searching in the
-        // current directory (the directory in which tsc was invoked).
-        // The server must start searching from the directory containing
-        // the newly opened file.
-        private findConfigFile(searchPath: NormalizedPath, projectRootPath?: NormalizedPath): NormalizedPath {
+        /**
+         * This function tries to search for a tsconfig.json for the given file.
+         * This is different from the method the compiler uses because
+         * the compiler can assume it will always start searching in the
+         * current directory (the directory in which tsc was invoked).
+         * The server must start searching from the directory containing
+         * the newly opened file.
+         */
+        private enumerateConfigFileLocations(info: ScriptInfo,

```

Comment:
Nit: I would usually expect a method like this to be named `forEachConfigFileLocation`.